includes/Utilities/Plugin_Request_Utility.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Utilities
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Utilities;
+
+use Exception;
+
+/**
+ * Class providing utility methods to return plugin information based on the request.
+ *
+ * @since n.e.x.t
+ */
+class Plugin_Request_Utility {
+
+	/**
+	 * Returns the plugin basename based on the input provided.
+	 *
+	 * @param string $plugin_slug The plugin slug or basename.
+	 * @return string The plugin basename.
+	 *
+	 * @throws Exception Thrown if an invalid basename or plugin slug is provided.
+	 */
+	public static function get_plugin_basename_from_input( $plugin_slug ) {
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+		if ( empty( $plugin_slug ) ) {
+			throw new Exception( 'Missing positional argument. Please provide the plugin slug as first positional argument.' );
```

Comment:
Please update the message here to be context-independent. This message worked well when it was directly scoped to the CLI command, but now referring to a "positional argument" no longer makes sense.

Same applies to the other messages below.
```suggestion
			throw new Exception( 'Invalid plugin slug: Plugin slug must not be empty.' );
```